src/lib.rs

```diff
@@ -395,6 +395,83 @@ impl<T: Ord> OrderedCollection<T> {
         // SAFETY: 1 <= i, so not [0], so initialized
         (i > 0).then(|| unsafe { self.items.get_unchecked(i).assume_init_ref() })
     }
+
+    /// Iterator over elements of a collection.
+    ///
+    /// It yields all items in an unspecified order.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use ordsearch::OrderedCollection;
+    /// let expected = vec![1, 2, 3, 4, 5];
+    /// let coll = OrderedCollection::from(expected.clone());
+    /// let mut values: Vec<_> = coll.iter().copied().collect();
+    /// values.sort();
+    /// assert_eq!(values, expected);
+    /// ```
+    pub fn iter(&self) -> impl Iterator<Item = &T> {
+        Iter { coll: self, idx: 0 }
+    }
+}
+
+impl<T: Clone> OrderedCollection<T> {
+    /// Copies all elemenets into a new [`Vec`] in unspecified order
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use ordsearch::OrderedCollection;
+    /// let x = vec![1, 2, 3, 4, 5];
+    /// let coll = OrderedCollection::from(x.clone());
+    /// let mut values: Vec<_> = coll.into_vec();
+    /// values.sort();
+    /// assert_eq!(values, x);
+    /// ```
+    pub fn into_vec(mut self) -> Vec<T> {
```

Comment:
I'm tempted to not have this method and _just_ have the `From`, but don't feel too strongly — what do you think?